### ISSUE TITLE
Add a route name field to each route in route.Route list (HTTP route)

### ISSUE THREAD

**Issue by drao9** [envoyproxy/envoy#6364]
**Issue Template**

*Title*:  Add a route name field to each route in route.Route list (HTTP route)

*Description*:
This is a feature request for adding a name field to the http routes. It would be enabled by default but if the name is not specified in the config, then it won't be enabled. This feature has helped us a lot with logging and debugging at eBay. One of the use cases is when there are 100s of routes and we want to check which route has been matched; the name makes it easier to identify. We wanted to see if this will be useful for the community. We have a PR that can be made for this case. If there is interest, I can also make a subsequent PR adding the route name field to access logs. Here is an example of how the config JSON for the route.Route object would look like: 

![Screen Shot 2019-03-22 at 4 19 36 PM](https://user-images.githubusercontent.com/12895455/54857962-5e88d100-4cbf-11e9-99af-40218b872e36.png)

**Comment by mattklein123** [comment]
@drao9 yes I think this would be very useful to a lot of people. (This has come up at Lyft also.) It could also be the basis for allowing optional per-route stats also.

**Comment by drao9** [comment]
I was out on vacation the past few weeks - I am working on this now and will have a PR in a couple of days.
